### PR TITLE
API: Endpoint for retrieving the list of the current events/study groups

### LINKED ISSUE #57: API: Endpoint for retrieving the list of the current events/study groups
Create an endpoint for returning a list of the current events/study groups and it's status.

image

Please take in account the date and time of the event and the max participants and the several status of the event/study group.

### PR DESCRIPTION
**Pull request by AnilMula**
# Description

This PR is to create end point for retrieving the list of current events

Fixes #57 

# How to test?

not tested

# Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/controllers/events.controller.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const getEventById = async (id) => {
+  try {
+    const events = await knex('events')
+      .select('events.id as id', 'event_type')
+      .where({ id });
+    if (events.length === 0) {
+      throw new Error(`incorrect entry with the id of ${id}`, 404);
+    }
+    return events;
```
Comment (tpetrina) [inline src/server/api/controllers/events.controller.js:20]: You should return just the one element that you find here, not an array. Have you tested with postman?

File: src/server/api/controllers/events.controller.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const knex = require('../../config/db');
+const Error = require('../lib/utils/http-error');
+
+const getEvents = async () => {
+  try {
+    return await knex('events').select('events.id', 'event.event_type');
```
Comment (tpetrina) [inline src/server/api/controllers/events.controller.js:6]: Please also return date, venue, description, and max_participants.

File: src/server/api/routes/events.router.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+ *
+ *            maxParticipants:
+ *              type: int
+ *    responses:
+ *      201:
+ *        description: event created
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.post('/', (req, res) => {
```
Comment (tpetrina) [inline src/server/api/routes/events.router.js:96]: If you added these routes, why didn't you implement them? Did you test if they work?

**Review by tpetrina — CHANGES_REQUESTED** [review]
Good start.

File: src/server/api/controllers/events.controller.js (hunk 1 above)
Comment (AnilMula) [reply src/server/api/controllers/events.controller.js]: modified

File: src/server/api/controllers/events.controller.js (hunk 2 above)
Comment (AnilMula) [reply src/server/api/controllers/events.controller.js]: added all these column names

**Review by tpetrina — CHANGES_REQUESTED** [review]
@AnilMula Have you tried running this code? It doesn't work. You need to fix the errors and test it either in the browser or via Postman (or similar tool).

File: src/server/api/routes/events.router.js (hunk 3 above)
Comment (tpetrina) [reply src/server/api/routes/events.router.js:96]: You still haven't removed these. They are breaking your code.

**Review by tpetrina — APPROVED** [review]
Good job!

**Review by Deepti-Dk — APPROVED** [review]

**Pull request merged**